submissions/yhorodechna/side-menu-dom/js/script.js

```diff
@@ -0,0 +1,142 @@
+        author: '— J.K Rowling, Harry Potter and the Philosophers Stone',
+        id: 'ravenclaw'
+    },
+    {
+        title: 'Slytherin',
+        img: 'img/slytherin_crest.png',
+        desc: 'The Slytherin house is the Hogwarts house that has sprung the darkest wizards. However, that does not mean that all Slytherins are dark wizards or evil, which will vary on a person-to-person basis. Slytherins house personality traits are ambition, leadership, self-preservation, cunning, and resourcefulness. Its founder, Salazar Slytherin, had a special hereditary trait, he was a Parselmouth. Parseltongue is the language of the serpents, and a wizard or witch that is a parselmouth is able to communicate with magical — yes, a basilisk — and no magical serpents.  Salazars connection to snakes was probably what influenced the Slytherin House symbol to be a serpent.  As mentioned in Helena Ravenclaws (aka. The Grey Lady) story above, The Baron was the one responsible for her death. Yet, what has not mentioned in the Ravenclaw portion of this article is that The Baron was a hot-tempered sort man madly in love with Helena Ravenclaw. After being asked by Rowena to go find her daughter and bring her back to Hogwarts, The Baron found Helena, but she refused to return to Hogwarts. In an episode of rage that ended up killing his loved one, and filled with remorse The Baron used the same weapon to take his own life. After his death, the Slytherin ghost was title The Bloody Baron since they were soaked in blood (his own blood, and Helena Ravenclaws).',
+        quote: '"Or perhaps in Slytherin youll make your real friends. Those cunning folk use any means to achieve their ends."',
+        author: '— J.K Rowling, Harry Potter and the Philosophers Stone',
+        id: 'slytherin'
+    }
+];
+
+const NAV_BAR_ITEMS = DATA.map(item => {
+    return {
+        title: item.title,
+        id: item.id
+    }
+});
+
+function createNavList(obj) {
+    let activeElId;
+    const { items,
+        parentEl,
+        defaultId,
+        navClick } = obj;
+    activeElId = defaultId;
+    const navElements = items.map((item) =>
+        `<li class="nav__item">
+            <a id="${item.id}" onclick2="navLinkClick(this,event)" href="#" class="nav__link ${defaultId === item.id ? 'active' : ''}">
+                <span class="nav__text">${item.title}</span> 
+            </a>
+        </li>`
+    );
+    const html = `
+        <header class="header">
+            <a class="header__link" onclick2="headerLinkClick()" ><img src="img/logo.png" alt="" class="img__btn"></a>
+            <h2 class="header__title">The Houses of Hogwarts</h2>
+        </header>
+        <nav class="nav"><ul class="nav__list">${navElements.join('')}</ul></nav>
+                `;
+    const div = document.createElement("div");
+    div.className = 'container';
+    div.innerHTML = html;
+    div.addEventListener("click", (event) => {
```

Comment:
for better readability, you can separate handlers from listeners

Reply:
Solved.